lib/std/Build/Cache.zig

```diff
@@ -765,7 +773,7 @@ pub const Manifest = struct {
         errdefer gpa.free(prefixed_path.sub_path);
 
         const gop = try self.files.getOrPutAdapted(gpa, prefixed_path, FilesAdapter{});
-        errdefer assert(self.files.popOrNull() != null);
+        errdefer _ = self.files.pop();
```

Comment:
FYI I plan to eventually remove `pop` and rename `popOrNull` to `pop`. This will become a standard Zig API pattern: have the version that provides the optional, and let the user do the `.?` if they want it.

Reply:
I only objected to actively discarding a value by comparing it with null, `errdefer _ = self.files.popOrNull().?;` has the same effect and the only reason not to write that is because `self.files.pop()` exists and does the same thing.

Reply:
I'm aware of this comment, but I merged this fix separately as #19417 anyway, since this issue is a pretty big one which causes frequent random crashes (and, worse, things like false test passes) so I figured it made sense to get the fix into the queue so that it's in nightly ASAP.